.Dislocation Motion.tmp.nlogo

```diff
@@ -275,104 +283,113 @@ to calculate-fl-positions
-  ; adjusting the forces to account for any external applied forces
-  let ex-force 0
-  if ex-force-applied? = 1 [ set ex-force report-new-force ]
-  if shape = "circle-x" and pinned? = 0 and ex-force-applied? = 0 [ set shape "circle" ]
-  set new-fx ex-force + new-fx
-
-  ; updating velocity and force
-  set vx velocity-verlet-velocity vx (fx / mass) (new-fx / mass)
-  set vy velocity-verlet-velocity vy (fy / mass) (new-fy / mass)
-  set fx new-fx
-  set fy new-fy
+  if not pinned? [
+    ; adjusting the forces to account for any external applied forces
+    let ex-force 0
+    if ex-force-applied? [ set ex-force report-new-force ]
+    if shape = "circle-dot" and not ex-force-applied? [ set shape "circle" ]
+    set new-fx ex-force + new-fx
+
+    ; updating velocity and force
+    set vx velocity-verlet-velocity vx (fx / mass) (new-fx / mass)
+    set vy velocity-verlet-velocity vy (fy / mass) (new-fy / mass)
+    set fx new-fx
+    set fy new-fy
+  ]
 
-  update-atom-color total-force
+  update-atom-color total-potential
   update-links in-radius-atoms
 end
 
 to update-atom-color [total-force] ; updating atom color
-  (ifelse update-color? [
+  (ifelse update-atom-color? [
     set-color total-force
   ]
    [ set color blue ])
 end
 
 to update-links [in-radius-atoms] ; updating links
-  if diagonal-right-links [
+  if show-diagonal-right-links? [
     set heading 330
     link-with-atoms-in-cone in-radius-atoms
   ]
-  if diagonal-left-links [
+  if show-diagonal-left-links? [
     set heading 30
     link-with-atoms-in-cone in-radius-atoms
   ]
-  if horizontal-links [
+  if show-horizontal-links? [
     set heading 90
     link-with-atoms-in-cone in-radius-atoms
   ]
 end
 
 to link-with-atoms-in-cone [atom-set]
-  let in-cone-atoms (atom-set in-cone cone-check-dist 60)
+  let in-cone-atoms (atom-set in-cone link-check-dist 60)
     if any? in-cone-atoms [
       create-atom-link-with min-one-of in-cone-atoms [distance myself]
     ]
 end
 
 to-report report-new-force
-  set shape "circle-x"
+  set shape "circle-dot"
   (ifelse force-mode = "Tension" [
-    report -1 * f-app-auto / num-forced-atoms
+    report -1 * f-app / num-forced-atoms
     ]
     [ ; Shear and Compression
       report f-app / num-forced-atoms
     ]
   )
 end
 
-to-report LJ-force [ r ] ; + = attract, - = repulse (this derivative would usually have a negative sign in front, so it's as if we multiplied it by a negative)
+to-report LJ-poten-and-force [ r ] ; + = attract, - = repulse (this derivative would usually have a negative sign in front, so it's as if we multiplied it by a negative)
   let third-pwr (sigma / r) ^ 3
   let sixth-pwr third-pwr ^ 2
-  report (48 * eps / r )* (sixth-pwr ^ 2 - (1 / 2) * sixth-pwr) + .0001
+  let twelfth-pwr sixth-pwr ^ 2
+  let force (48 * eps / r ) * (twelfth-pwr - (1 / 2) * sixth-pwr) + .0001
   ;report (48 * eps / r )* ((sigma / r) ^ 12 - (1 / 2) * (sigma / r) ^ 6) + .0001
+  let potential (4 * eps * (twelfth-pwr - sixth-pwr))
```

Comment:
The potential should also have an adjustment factor so that it is zero at the cutoff distance.